src/troubleshooting/performance.rst

```diff
@@ -0,0 +1,88 @@
+Performance tuning in SSSD
+##########################
+
+Slow id lookup
+**************
+
+It has been noticed that the id lookup becomes slow if the LDAP or AD user is a member of large groups (viz., 500+ groups). ``id`` is very heavy. ``id`` does a lot under its hood.
+Behind the scenes, when the ``id $user`` command is executed, it triggers the following:
+ 
+- Get user information - getpwnam() for the user.
+ 
+- Get primary group information - getgrgid() for the primary group of the user.
+ 
+- Get list of groups - getgrouplist() for the user.
+ 
+- Get group information for each group returned from step 3 - getgrid() for all GIDs returned from the getgrouplist() call.
```

Comment:
Steps above should be numbered, as you've listed step numbers in further comments below.